laurent22/joplin
# Android: Toggle plugin panels using a button in the toolbar

# Summary

Adjusts the UI for showing the plugin panels dialog.

See [the relevant post on the forum](https://discourse.joplinapp.org/t/mobile-plugins-best-way-to-support-opening-plugin-panels/36892).

# Notes

- The plugin panel toggle is only shown if a plugin is installed that registers a plugin panel.
- This decreases the available space for the notebook selector dropdown.
    - I plan to create a follow-up pull request that hides the plugin panel button (and perhaps other buttons) when opening the notebook dropdown ([draft implementation](https://github.com/personalizedrefrigerator/joplin/compare/38edd17907b32eb2ab41e8261a52efad170c827a..bd0d1a1dcd31b7ef1adadee94dbdc9dbf989d52c)).
- Several bugs related to [PluginPanelViewer.tsx](https://github.com/laurent22/joplin/compare/dev...personalizedrefrigerator:joplin:pr/mobile-plugin-support/plugin-panels-toggle-in-toolbar#diff-3966b11d9083ad2a8ea08050816b438e0af882aed86c16433c9b7e8395718c6e) being openable from the settings screen (where plugins can be enabled/disabled) have been fixed.


# Screenshots

![screenshot: Puzzle icon (plugin icon) is just to the left of the magnifying glass (search) in the toolbar, note list visible](https://github.com/laurent22/joplin/assets/46334387/4ab1e3b1-874a-4538-aeb2-7aba0c863925)
![screenshot: History panel from the history plugin is visible](https://github.com/laurent22/joplin/assets/46334387/497a341e-598c-45d5-90c7-09f5ee9c2322)
![screenshot: Plugin icon is shown taking up space in the toolbar, the notebook dropdown is small](https://github.com/laurent22/joplin/assets/46334387/94ba5f2a-b23a-45c3-8218-233f4b492ac9)
![screenshot: The plugin icon is visible in the settings screen, in a light theme](https://github.com/laurent22/joplin/assets/46334387/3193d930-5479-46a4-af09-75b22896ea4f)

# Testing

1. Uninstall the History Panel and Outline plugins if already installed
2. Verify that no plugin icon is visible in the toolbar
3. Install the History Panel plugin
4. Install the Outline plugin
5. Verify that the plugin icon is visible in the toolbar
6. Click the plugin icon
7. Verify that both the history and outline tabs are visible
8. Use the history panel to navigate to a note
9. Close the history panel and edit the note
10. Open the history panel using the plugin button
11. Switch to the outline tab
12. Close the panel
13. Enable TalkBack
14. Open the panel dialog using TalkBack
    - Verify that focus moves to the dialog (the close button)
15. Navigate to the tab selector
16. Switch tabs
17. Verify that the name of the new tab is announced
18. Close the dialog with TalkBack

This has been tested successfully on an Android 12 emulator.
